doc/developers.rst

```diff
@@ -934,12 +948,12 @@ initial schedule. There are three cases:
 1) loops that iterate over cells and modify a continuous field will
    access the level-1 halo. This means that any field that is read
    within such a loop must have its level-1 halo clean and therefore
-   requires a halo exchange. Note, at the moment PSyclone adds a halo
-   exchange for the modified field (as it is specified as `GH_INC`
-   which requires a read before a write), however this is not required
-   if there is only one field updated in the kernel. This is because
-   we only care about updating owned and annexed dofs, therefore it
-   does not matter what the values of any halo dofs are.
+   requires a halo exchange. In the case of a modified field
+   (specified as `GH_INC` which involves a read before a write) will
+   require a halo exchange if its annexed dofs are not up-to-date, or
```

Comment:
Dropped the "In the case of" and rewrote the explanation as to why halo exchanges need to be added. Hopefully the changes will address your concerns. For the record, it would require a halo exchange if its annexed dofs were dirty.